sys/x86/x86/local_apic.c

```diff
@@ -1272,6 +1272,9 @@ lapic_handle_intr(int vector, struct trapframe *frame)
 {
 	struct intsrc *isrc;
 
+	critical_enter();
+	++curthread->td_intr_nesting_level;
+
 	kasan_mark(frame, sizeof(*frame), sizeof(*frame), 0);
 	kmsan_mark(&vector, sizeof(vector), KMSAN_STATE_INITED);
 	kmsan_mark(frame, sizeof(*frame), KMSAN_STATE_INITED);
```

Comment:
I'm unsure of the ordering here.  Should these instead be in the opposite order?  (similar issue above in `atpid.c`)